docs/system/mx4/power_management.md

```diff
@@ -0,0 +1,210 @@
+#define WAKE_UP_SRC_DIN_7_R         0x25
+#define WAKE_UP_SRC_DIN_8_F         0x26
+#define WAKE_UP_SRC_DIN_8_R         0x27
+```
+
+##### Wake on Analog Inputs
+
+Analog inputs as wakeup sources are not managed by `go_to_sleep.sh`.
+
+All analog inputs have four sysfs files associated with them. For example, if we take input voltage:
+
+[Sysfs files and their purposes]
+
+- `input_voltage` reads the value of input voltage.
+- `input_voltage_calibration_u` calibrates the input for component tolerances.
+- `input_voltage_threshold_high` enables wake on a high-level threshold.
+- `input_voltage_threshold_low` enables wake on a low-level threshold.
+
+Examples for wakeup setup:
+
+```bash
+# Wakeup system from sleep/suspend if input voltage is above 16 V
+root@mx4-vcc-1000000:~# echo 16000 > /opt/hm/pic_attributes/input_voltage_threshold_high
+```
+
+```bash
+# Wakeup system from sleep/suspend if input voltage is below 12 V
+root@mx4-vcc-1000000:~# echo 12000 > /opt/hm/pic_attributes/input_voltage_threshold_low
+```
+
+```bash
+# Wakeup system from sleep/suspend if input voltage is in the range of 12-16 V
+root@mx4-vcc-1000000:~# echo 16000 > /opt/hm/pic_attributes/input_voltage_threshold_high
+root@mx4-vcc-1000000:~# echo 12000 > /opt/hm/pic_attributes/input_voltage_threshold_low
+```
+
+```bash
+# Write 0 to both threshold files to disable that specific input as wakeup source
+root@mx4-vcc-1000000:~# echo 0 > /opt/hm/pic_attributes/input_voltage_threshold_high
+root@mx4-vcc-1000000:~# echo 0 > /opt/hm/pic_attributes/input_voltage_threshold_low
+```
+User setup for analog wakeup sources is required before calling `go_to_sleep.sh`. See [Enter sleep](#enter-sleep).
+
+##### Wake on CAN
+
+Wake on CAN isn't supported on all MX-4 platforms. Contact Host Mobility support for compatibility information.
+
+Enable Wake on CAN by passing `-c` to `go_to_sleep.sh` (see [Enter sleep](#enter-sleep)). However, before doing this, specific CAN buses should be enabled as wakeup sources.
+
+By default, all CAN buses will trigger a wakeup if there is traffic on them and `-c` is passed to `go_to_sleep.sh`. For T30 FR it is possible to wake up on single specific CAN frame(`can-xcvr`).
+
+GPIOs are used to set this up:
+
+```bash
+root@mx4-vcc-1000000:~# cat /sys/kernel/debug/gpio | grep -i wakeup
+ gpio-58  (P169 - CAN0-WAKEUP  ) out lo
+ gpio-59  (P171 - CAN1-WAKEUP  ) out lo
+ gpio-60  (P173 - CAN2-WAKEUP  ) out lo
+ gpio-61  (P175 - CAN3-WAKEUP  ) out lo
+ gpio-62  (P177 - CAN4-WAKEUP  ) out lo
+ gpio-63  (P179 - CAN5-WAKEUP  ) out lo
+```
+
+To disable wakeup for a specific bus, write a 1 to that GPIO's value.
+
+```bash
+# Disable wakeup on CAN0
+root@mx4-vcc-1000000:~# echo 1 > /sys/class/gpio/gpio58/value
+
+```
+
+### Deep Sleep
+
+Deep sleep mode puts the co-CPU to sleep while cutting the power rail to the main CPU. Upon wakeup, a cold reboot occurs instead of a fast resume, trading off for lower consumption during the suspended state.
```

Comment:
```suggestion
Deep sleep mode puts the coprocessor to sleep while cutting the power rail to the main CPU. Upon wakeup, a cold reboot occurs instead of a fast resume, trading off for lower consumption during the suspended state.
```